### DISCUSSION TITLE
Batch apply alpha mask

### DISCUSSION THREAD

**Discussion by kolyafromburgerking** [ImageMagick/ImageMagick#7479] [Help] [answered]
### ImageMagick version

7.1.1-27

### Operating system, version and so on

Windows 10

### Description

I have a lot of BMP images that consist of two halves: one on the left is the colored image with a #00FF00 background, and on the right is the mask, similar to this:
![alice](https://kaisernet.org/onscripter/dev/alice.jpg)
I want to apply the mask to the colored image so that the lightness of the right image corresponds to the alpha channel, and crop out the mask, as well as convert to PNG.

I need to do this with thousands of files at once. Sorry if this is a basic question, I'm a beginner.

I found someone that needed to do the same thing as me, but they used magick++ ([link](https://stackoverflow.com/questions/29915460/applying-alpha-mask-to-image-in-magick)), is there a way to convert their magick++ code to magick command line?

**With snibgo's help, this is the command I ended up using:**
```
mogrify -format png *.bmp
FOR %a in (*.png) DO magick %a -crop 2x1@ +repage ( -clone 1 -negate ) -delete 1 -alpha off -compose CopyOpacity -composite %a
```

**Comment by snibgo** [comment]
> I managed to find out how to do this with one image, ...

You can either change that command to `magick mogrify`, or put your command into a shell `for` loop.

**Reply by kolyafromburgerking** [reply]
Sorry, I got confused. I found out how to do this if the left image has a black background, like in the example, but not with a green background, so I need help with that as well.

**Reply by kolyafromburgerking** [reply]
i found someone that needed to do the same thing as me, but they used magick++ ([link](https://stackoverflow.com/questions/29915460/applying-alpha-mask-to-image-in-magick)), is there a way to convert their magick++ code to magick command line?

**Comment by snibgo** [comment, marked as answer]
> is there a way to convert their magick++ code to magick command line?

Yes, but I would do it slightly differently, like this:
```
%IMG7%magick ^
  girlMask.jpg ^
  -crop 2x1@ +repage ^
  ( -clone 1 ^
    -negate ^
  ) ^
  -delete 1 ^
  -alpha off ^
  -compose CopyOpacity -composite ^
  out.png
```
... where `girlMask.jpg` is my name for the file you showed. Obviously, use the name of your BMP file, not a JPG.

> I found out how to do this if the left image has a black background, like in the example, but not with a green background, so I need help with that as well.

What difference does the background of the left side make?

**Reply by kolyafromburgerking** [reply]
thank you, this worked. i guess there isn't any difference in the background after all
